Improve checks and rule speed, add JSON support

tystuyfzand opened the pull request:
- Clears cache on server state change (prevents no servers being available)
- Improves speed of GetValue using generated getters
- Moves maxminddb items into db package
- Moves util.go into util package
- Updates dependencies

tystuyfzand committed:
[feature] Add JSON map support

- Adds JSON map file support

tystuyfzand committed:
[feature] Improve speed of GetValue, clear cache on server change

- Clears cache on server state change (prevents no servers being available)
- Improves speed of GetValue using generated getters
- Moves maxminddb items into db package
- Moves util.go into util package
- Updates dependencies

efectn commented on map_test.go:
```diff
@@ -7,8 +7,8 @@ import (
-	It("Should successfully load the map", func() {
-		m, err := loadMap(strings.NewReader(`bananapi/Bullseye_current|bananapi/archive/Armbian_21.08.1_Bananapi_bullseye_current_5.10.60.img.xz|Aug 26 2021|332M`))
+	It("Should successfully load the map from a CSV/Pipe separated file", func() {
+		m, err := loadMapCSV(strings.NewReader(`bananapi/Bullseye_current|bananapi/archive/Armbian_21.08.1_Bananapi_bullseye_current_5.10.60.img.xz|Aug 26 2021|332M`))
```
Can you add a testcase for JSON, too

tystuyfzand replied:
It's super super basic, but I've added one. It should probably have a proper failure case/etc at some point, but it confirms that it works properly.

tystuyfzand committed:
[bugfix] Fix compile errors

- Fix compile errors not found by prechecks

tystuyfzand committed:
[bugfix] Correct format of JSON

- Corrects format of JSON by adding assets parent
- Fixes FileURL variable to map to the proper field
- Adds basic unit test for JSON loading
